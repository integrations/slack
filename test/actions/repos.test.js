const repos = require('../../lib/actions/repos');

describe('Test Repo List for Action Message', () => {
  test('get', async () => {
    const res = {
      send: result => result,
      locals: {
        gitHubUser: {
          client: {
            users: {
              get: () => ({
                data: {
                  login: 'userName',
                },
              }),
            },
            repos: {
              getForUser: () => ({
                data: [
                  {
                    id: 123,
                    full_name: 'awesome/repo',
                  },
                  {
                    id: 321,
                    full_name: 'awesome/repo2',
                  },
                ],
              }),
            },
          },
        },
      },
    };

    const req = {};
    const result = await repos.get(req, res);

    expect(result).toMatchSnapshot();
  });
});
