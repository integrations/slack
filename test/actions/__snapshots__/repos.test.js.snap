// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Repo List for Action Message get 1`] = `
Object {
  "options": Array [
    Object {
      "label": "awesome/repo",
      "value": 123,
    },
    Object {
      "label": "awesome/repo2",
      "value": 321,
    },
  ],
}
`;
